### PR TITLE
Supporting core

### PR DESCRIPTION
**Pull request by 20shivangi**

### CODE REVIEW

File: Tasks/AzurePowerShellV4/Tests/RemovesFunctionsAndVariables.ps1 [hunk 1]
```diff
@@ -15,7 +15,8 @@ Register-Mock Update-PSModulePathForHostedAgent
 Register-Mock Remove-EndpointSecrets
```
[the remark this answers is not in the document]
Comment (20shivangi) [reply Tasks/AzurePowerShellV4/Tests/RemovesFunctionsAndVariables.ps1]: done

File: Tasks/AzurePowerShellV4/AzurePowerShell.ps1 [hunk 2]
```diff
@@ -48,109 +50,115 @@ if ($targetAzurePs -eq $latestVersion) {
...
+        $__vstsAzPSScriptPath,
+        $joinedContents,
+        ([System.Text.Encoding]::UTF8))
+
+    # Prepare the external command values.
+    #
+    # Note, use "-Command" instead of "-File". On PowerShell v4 and V3 when using "-File", terminating
+    # errors do not cause a non-zero exit code.
+    if ($input_pwsh) {
+        $powershellPath = Get-Command -Name pwsh.exe -CommandType Application | Select-Object -First 1 -ExpandProperty Path
```
[the remark this answers is not in the document]
Comment (20shivangi) [reply Tasks/AzurePowerShellV4/AzurePowerShell.ps1]: done

File: Tasks/AzurePowerShellV4/AzurePowerShell.ps1 [hunk 3]
```diff
@@ -10,6 +10,8 @@ $__vsts_input_errorActionPreference = Get-VstsInput -Name errorActionPreference
 $__vsts_input_failOnStandardError = Get-VstsInput -Name FailOnStandardError
 $targetAzurePs = Get-VstsInput -Name TargetAzurePs
 $customTargetAzurePs = Get-VstsInput -Name CustomTargetAzurePs
+$input_pwsh = Get-VstsInput -Name 'pwsh'
```
Comment (niadak) [inline Tasks/AzurePowerShellV4/AzurePowerShell.ps1]: Why this is hard coded?

Comment (niadak) [inline Tasks/AzurePowerShellV4/AzurePowerShell.ps1]: Please validate the task before check in.

File: Tasks/AzurePowerShellV4/Tests/ChecksForPowerShellCore.ps1 [hunk 4]
```diff
@@ -0,0 +1,28 @@
...
+Register-Mock Remove-EndpointSecrets
+Register-Mock Disconnect-AzureAndClearContext
+Register-Mock Assert-VstsPath
+Register-Mock Invoke-VstsTool { }
+
+# Act.
+$actual = & $PSScriptRoot\..\AzurePowerShell.ps1
+
+Assert-WasCalled Invoke-VstsTool -Times 1
+Assert-WasCalled Invoke-VstsTool -ArgumentsEvaluator {$args.count -eq 6 -and $args[3] -like '*pwsh.exe*'}
```
Comment (niadak) [inline Tasks/AzurePowerShellV4/Tests/ChecksForPowerShellCore.ps1]: Can you try this. $args | ConvertTo-Json) -like "*powershell.exe*"

File: Tasks/AzurePowerShellV4/AzurePowerShell.ps1 (hunk 3 above)
Comment (20shivangi) [reply Tasks/AzurePowerShellV4/AzurePowerShell.ps1]: done

File: Tasks/AzurePowerShellV4/Tests/ChecksForPowerShellCore.ps1 (hunk 4 above)
Comment (20shivangi) [reply Tasks/AzurePowerShellV4/Tests/ChecksForPowerShellCore.ps1]: done

File: Tasks/AzurePowerShellV4/AzurePowerShell.ps1 (hunk 3 above)
Comment (20shivangi) [reply Tasks/AzurePowerShellV4/AzurePowerShell.ps1]: Done

**Comment by y325A** [comment]
@20shivangi thanks for this commit - it was something I really needed!

Do you know how long it takes for new versions of tasks to be released to Azure Devops?

**Comment by 20shivangi** [comment]
@y325A It will take at least 3-4 weeks to get deployed.

**Comment by brwilkinson** [comment]
This change now seems to have rolled out and I believe introduced several bugs... do those bugs also take a month to roll into a new version on AzDevOps ?

**Comment by brwilkinson** [comment]
okay all good now #11635 has resolved for me.

**Pull request merged**